### PR TITLE
[MINOR][PYTHON] Fix `MultilayerPerceptronClassifierTest.test_raw_and_probability_prediction`

### PR DESCRIPTION
**Pull request by harupy**
### What changes were proposed in this pull request?

- A follow-up for https://github.com/apache/spark/pull/35778
- Increase `rtol` to de-flake the test.

https://github.com/apache/spark/runs/5725350556

```
Not equal to tolerance rtol=0.15, atol=1e-08

Mismatched elements: 3 / 3 (100%)
Max absolute difference: 4.39314499
Max relative difference: 0.23356037
 x: array([-14.190332, -10.063731,  26.570715])
 y: array([-11.608192,  -8.15828 ,  22.17757 ])
```

### CODE REVIEW

File: python/pyspark/ml/tests/test_algorithms.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def test_raw_and_probability_prediction(self):
         np.testing.assert_allclose(
             result.rawPrediction,
             expected_rawPrediction,
-            rtol=0.15,
+            rtol=0.3,
```
Comment (harupy) [inline python/pyspark/ml/tests/test_algorithms.py:109]: Test

```python
import numpy as np

a = np.array([-14.190332, -10.063731, 26.570715])
b = np.array([-11.608192, -8.15828, 22.17757])

atol = 1e-08
print(np.max((np.abs(a - b) - atol) / np.abs(b)))  # 0.2335603816000433

rtol = 0.3
np.testing.assert_allclose(a, b, rtol=rtol, atol=rtol)
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
im gonna merge this first, and montior if the test is still flaky.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.3.

**Comment by dongjoon-hyun** [comment]
Thank you, @harupy and @HyukjinKwon .

**Pull request closed without merging**